### DISCUSSION TITLE
Storing a reference to a websocket client (4.0.0)

### DISCUSSION THREAD

**Discussion by uellenberg** [actix/actix-web#2622] [Q&A]
Hello, I'm currently designing a system using websockets and I'd like to be able to send a message to one of the clients connected to my websocket server. I know how to send a response on a request, but I can't figure out how to store the client in a safe way and then send a message to them unprompted. I'm a bit new to Rust's way of memory management, so hopefully I'm not missing anything too obvious. Any help is appreciated!

**Comment by uellenberg** [comment]
I've been looking into using the address of the context send on a message to store it (`ctx.address()`). I've yet to test if this works though.

**Comment by bergkvist** [comment]
The answer I just got here might be helpful: https://github.com/actix/actix-web/discussions/2652

Essentially, if you are able to make the websockets actors themselves decide what to subscribe to (instead of the other way around), you can use async_streams to push values to websocket actors.
